code/game/objects/items/aiming.dm

```diff
@@ -0,0 +1,212 @@
+/obj/item/gun/Initialize()
+	. = ..()
+	AddComponent(/datum/component/aiming)
+
+/obj/item/reagent_containers/food/snacks/grown/banana/Initialize() //Yes, you can hold someone at gunpoint with a banana.
+	. = ..()
+	AddComponent(/datum/component/aiming)
+
+/mob/living/proc/aim_react()
+	set waitfor = FALSE
+	var/list/options = list()
+	for(var/option in list("surrender", "ignore"))
+		options[option] = image(icon = 'icons/effects/aiming.dmi', icon_state = option)
+	var/choice = show_radial_menu(src, src, options)
+	if(choice == "surrender")
+		emote("surrender")
+
+/datum/action/item_action/aim
+	name = "Manually aim"
+	desc = "Aim at a target manually. You can also aim with 'point' or SHIFT + middle mouse while holding a gun."
+	button_icon_state = "aim"
+	icon_icon = 'icons/mob/actions/actions_security.dmi'
+
+/obj/item/ui_action_click(mob/user, action)
+	if(istype(action, /datum/action/item_action/aim))
+		var/datum/component/aiming/aiming = GetComponent(/datum/component/aiming)
+		if(!aiming)
+			return
+		var/list/mobs = list()
+		for(var/mob/living/M in view(user, 15))
+			if(M == user || !isliving(M) || M.invisibility > 0 || !M.alpha)
+				continue
+			mobs += M
+		var/mob/living/A = input(user, "Who do you want to point [src] at?", "[src]", null) as null|anything in mobs
+		if(isliving(A))
+			aiming.aim(user, A)
+		return
+	. = ..()
+
+//Happens before the actual projectile creation
+/obj/item/gun/before_firing(atom/target,mob/user, aimed)
+	if(aimed)
+		if(chambered?.BB && !istype(src, /obj/item/gun/ballistic/automatic/toy))
```

Comment:
speed or stamina damage boost, choose one. Otherwise, this is just "middle click 2 do more damage lol". Hell, even with that it is

Reply:
Why does the bullet go faster if you aim, and 55 stamina damage is a lot of damage.

Reply:
By the time you click the "FIRE" button they'll likely be at the edge of your screen and thus easily able to dodge. By aiming + firing you'll still do less damage than you could have by just firing twice. It's so you don't get a heavy mechanical disadvantage by going for the roleplay-friendly route.